### DISCUSSION TITLE
Adding Azure icons

### DISCUSSION THREAD

**Discussion by msisolak** [likec4/likec4#861] [Ideas]
The new built-in icon sets are great. We work in Azure and I'd be happy to submit a PR to add their icons, but I'm not sure if we can. Here is the license text Microsoft provides:

> Azure architecture icons
Helping our customers design and architect new solutions is core to the Azure Architecture Center’s
mission. Architecture diagrams like those included in our guidance can help communicate design
decisions and the relationships between components of a given workload. On this page you will find an
official collection of Azure architecture icons including Azure product icons to help you build a custom
architecture diagram for your next solution.
Do’s
• Use the icons to illustrate how products can work together
• In diagrams, we recommend to include the product name somewhere close to the icon
• Use the icons as they would appear within Azure
Don’ts
• Don’t crop, flip or rotate icons
• Don’t distort or change icon shape in any way
• Don’t use Microsoft product icons to represent your product or service
Terms
Microsoft permits the use of these icons in architectural diagrams, training materials, or documentation.
You may copy, distribute, and display the icons only for the permitted use unless granted explicit
permission by Microsoft. Microsoft reserves all other rights.

Based on this, do you feel the Azure icons could be included as an embedded set in likec4?

**Comment by davydkov** [comment]
Hey @msisolak 

I think this is a good idea, and there seem to be no issues with embedding (like here [mingrammer/diagrams#980](https://github.com/mingrammer/diagrams/pull/980)).

I would appreciate it if you contribute. Here are the steps:
- add to [packages/icons/scripts/update-icons.mjs](https://github.com/likec4/likec4/blob/main/packages/icons/scripts/update-icons.mjs)
- update [packages/icons/package.json](https://github.com/likec4/likec4/blob/main/packages/icons/package.json)
- update [like-c4.langium](https://github.com/likec4/likec4/blob/a799d6a27911274ffa53f9cccf01b401c05b59ad/packages/language-server/src/like-c4.langium#L450)
- update [packages/likec4/scripts/prepack.ts](https://github.com/likec4/likec4/blob/main/packages/likec4/scripts/prepack.ts)
- update [packages/likec4/package.json](https://github.com/likec4/likec4/blob/main/packages/likec4/package.json)
- mention in docs [docs/dsl/styling.mdx](https://github.com/likec4/likec4/blob/a799d6a27911274ffa53f9cccf01b401c05b59ad/apps/docs/src/content/docs/dsl/styling.mdx?plain=1#L185)

Let me know if you have any issues.

**Comment by davydkov** [comment]
Planned  #1027
